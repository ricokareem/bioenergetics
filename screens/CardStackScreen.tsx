import React from "react";
import {
  FlatList,
  ListRenderItemInfo,
  Route,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import { Movies } from "../constants/MediaData";

type NavPropsType = {
  navigation: Route;
};

interface ItemType extends ListRenderItemInfo<any> {
  description: string;
  playlist: number[];
  subheading: string;
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#fff",
    flex: 1,
  },
  listItem: {
    backgroundColor: "#fc9829",
    flex: 1,
    flexDirection: "column",
    height: 120,
    justifyContent: "center",
    margin: 1,
    padding: 20,
  },
  listText: {
    color: "#fff",
    fontSize: 12,
    fontWeight: "bold",
    lineHeight: 14,
  },
  listSubheadingText: {
    color: "#fff",
    fontSize: 10,
    lineHeight: 12,
    marginTop: 5,
  },
});

const cardStack = [...Movies].filter((c) => c.playlist && c.playlist.length);

const CardStackScreen: React.FC<NavPropsType> = (props) => {
  const { navigation } = props;

  return (
    <View style={styles.container}>
      <FlatList
        data={cardStack}
        renderItem={({ item }: ItemType) => (
          <TouchableOpacity
            style={styles.listItem}
            testID="cardStackItem"
            onPress={() => {
              navigation.navigate(item.screen, {
                title: item.title,
                name: item.name,
                description: item.description,
                playlist: item.playlist,
              });
            }}
          >
            <Text style={styles.listText}>{item.title}</Text>
            <Text style={styles.listSubheadingText}>{item.subheading}</Text>
          </TouchableOpacity>
        )}
        numColumns={2}
        keyExtractor={(item, index) => `${index}_${item.name}`}
      />
    </View>
  );
};

export default CardStackScreen;
