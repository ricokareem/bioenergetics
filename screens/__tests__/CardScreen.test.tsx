import React from "react";
import { shallow } from "enzyme";
import CardStackScreen from "../CardStackScreen";

describe("<CardStackScreen>", () => {
  it("should match to snapshot", () => {
    const wrapper = shallow(<CardStackScreen navigation={{ id: "test" }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
