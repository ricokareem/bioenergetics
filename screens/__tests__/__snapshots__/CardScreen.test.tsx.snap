// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardStackScreen> should match to snapshot 1`] = `ShallowWrapper {}`;
